crates/sui-types/src/sui_system_state/sui_system_state_inner_v1.rs

```diff
@@ -40,6 +40,9 @@ pub struct SystemParametersV1 {
     /// The starting epoch in which stake subsidies start being paid out
     pub stake_subsidy_start_epoch: u64,
 
+    /// Minimum number of active validators at any moment.
+    pub min_validator_count: u64,
```

Comment:
I don't think this parameter was there in v1. @emmazzz might know for sure.
In anyway we should never modify older versions

Reply:
You are right, it wasn't. I'll check if I can only add it to `v2`.

Reply:
However, the larger issue is that the `min_validator_count` fields seems to be needed in the `GenesisChainParameters` and given that I never touched those parts of the codebase, it sounds a bit scary to mess with that.

Reply:
Right it's only added in `SystemParametersV2`.

Reply:
Upon further investigation, to get this field working we need it in `SuiSystemStateSummary`, which corresponds to a field's type that is needed to create `SuiSystemStateInnerV1`, which would break the rule of not modifying older versions. 
@amnn @oxade , how do we want to handle this field in the schema?

Reply:
Correct me if I'm misunderstanding here. We just need to add `min_validator_count` to `SuiSystemStateSummary` right? And the issue is what to do when we convert a `SuiSystemStateInnerV1` into `SuiSystemStateSummary` in `sui_system_state_inner_v1::into_sui_system_state_summary`? I think you can just give this field an arbitrary, default value such as 4.

Reply:
Right, we can do that. But isn't that violating the idea that V1 did not have this field? (I tried to do this and it gets more nested, genesis params needs to be handled, got some errors from the Move VM, etc.)

Reply:
can we build SuiSystemStateSummary from v2?

Reply:
I believe so, but the `v2` is internal, and the exposed interface with all the data was `SuiSystemStateSummary` so the design is not clear to me.

Reply:
Rn we use this function
```
    pub fn get_latest_sui_system_state(&self) -> Result<SuiSystemStateSummary, IndexerError> {
        let system_state: SuiSystemStateSummary =
            sui_types::sui_system_state::get_sui_system_state(self)?
                .into_sui_system_state_summary();
        Ok(system_state)
    }
```
We can just use the get_sui_system_state() part without the .into_sui_system_state_summary() to work with SuiSystemStateInner directly

Reply:
That could work nicely, let me try just for now and put a comment in there.

Reply:
I think @lxfind will agree that it's better to work with summary rather than the inner - in fact the whole reason why we have the summary type is so that we don't need to worry when the inner type gets upgraded to V2, V3, V4. And the "not touching older version" principal applies to the struct definition of V1 struct, not the `into_system_state_summary` method so providing an arbitrary value there should be good enough.

Reply:
I see, so SuiSystemStateSummary is fair game to extend then? We can just add a new field min_validator_count there?

Reply:
Yup!

Reply:
Thanks, I'll ping you for a review when it's ready.

Reply:
@emmazzz thanks for your patience. Would appreciate any feedback you might have. I have used the `MIN_VALIDATOR_COUNT` to give a value to `min_validator_count` field when converting from `inner_v1` to `SuiSystemStateSummary`, hope that's ok.